opentracing-api/src/main/java/io/opentracing/tag/Tag.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2016-2018 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.tag;
+
+import io.opentracing.Span;
+
+public interface Tag<T> {
+    String getKey();
```

Comment:
How about setValue(Span, T) instead?

Reply:
Had considered the same, but was afraid of making `AbstractTag.set()` public (right now is protected) ;)

Although now I think we could leave `set()` as abstract and have `setValue()`, as an alternative ;) Would that work for you?

Reply:
I would rather make set() public. I don't know why it's protected now, since all concrete classes make it public.